Core/automation/lib/python/core/date.py

```diff
@@ -269,6 +273,12 @@ def to_java_zoneddatetime(value):
     # Eclipse Smarthome DateTimeType
     if EclipseDateTimeType and isinstance(value, EclipseDateTimeType):
         return to_java_zoneddatetime(value.calendar)
+    # Eclipse Smarthome DateTimeItem
+    if EclipseDateTimeItem and isinstance(value, EclipseDateTimeItem):
+        return to_java_zoneddatetime(value.calendar)
```

Comment:
This should be:

```python
        return to_java_zoneddatetime(value.getState())
```

Reply:
Good catch! Sorry, I didn't have a OH2 instance to test this.